aexpect/client.py

```diff
@@ -193,9 +193,16 @@ def __init__(self, command=None, a_id=None, auto_close=False, echo=False,
             sub.stdin.write(f"{command}\n".encode(self.encoding))
             sub.stdin.flush()
             # Wait for the server to complete its initialization
-            while (f"Server {self.a_id} ready" not in
-                   sub.stdout.readline().decode(self.encoding, "ignore")):
-                pass
+            full_ouput = ""
+            pattern = f"Server {self.a_id} ready"
+            while True:
+                output = sub.stdout.readline().decode(self.encoding, "ignore")
+                if pattern in output:
+                    break
+                full_ouput += output
+                sub_status = sub.poll()
+                if sub_status is not None:
+                    raise ExpectProcessTerminatedError(pattern, sub_status, full_ouput)
```

Comment:
Thanks, this does make a lot of sense. I'm wondering whether we might want to add a timeout as well (to interrupt either when the process terminates or when it won't start in let's say 1 minute? Or 5 minutes to be sure?) What do you think?

Reply:
Good catch! I also considered this, but kept it the first time. I will add a timeout next update.